screenpy/actions/quietly.py

```diff
@@ -0,0 +1,117 @@
+"""
+Quietly allows for "disabling" logging on successful actions & tests
+"""
+from __future__ import annotations
+
+from typing import Any, TypeVar, Union, overload
+
+from hamcrest.core.base_matcher import Matcher
+from typing_extensions import TypeAlias
+
+from screenpy import settings
+from screenpy.actor import Actor
+from screenpy.pacing import the_narrator
+from screenpy.protocols import Answerable, Performable, Resolvable
+
+T = TypeVar("T")
+
+
+class QuietlyMixin:
+    """
+    Quietly needs to mimic the ducks (i.e. objects) they are wrapping.
+    All of the attributes from the "duck" should be exposed by the Quietly object.
+    """
+
+    def __getattr__(self, key: Any) -> Any:
+        try:
+            return getattr(self.duck, key)
+        except AttributeError as exc:
+            msg = (
+                f"{self.__class__.__name__}({self.duck.__class__.__name__}) "
+                f"has no attribute '{key}'"
+            )
+            raise AttributeError(msg) from exc
+
+
+class QuietlyPerformable(Performable, QuietlyMixin):
+    """Act like the Performable passed in but kink the cable when performing"""
+
+    def perform_as(self, actor: Actor) -> None:
+        with the_narrator.mic_cable_kinked():
+            self.duck.perform_as(actor)
+            if not settings.DEBUG_QUIETLY:
+                the_narrator.clear_backup()
+            return
+
+    def __init__(self, duck: Performable):
+        self.duck = duck
+
+
+class QuietlyAnswerable(Answerable, QuietlyMixin):
+    """Act like the Answerable passed in but kink the cable when answering"""
+
+    def answered_by(self, actor: Actor) -> Any:
+        with the_narrator.mic_cable_kinked():
+            thing = self.duck.answered_by(actor)
+            if not settings.DEBUG_QUIETLY:
+                the_narrator.clear_backup()
+            return thing
+
+    def __init__(self, duck: Answerable):
+        self.duck = duck
+
+
+class QuietlyResolvable(Resolvable, QuietlyMixin):
+    """Act like the Resolvable passed in but kink the cable when resolving"""
+
+    def resolve(self) -> Matcher:
+        with the_narrator.mic_cable_kinked():
+            res = self.duck.resolve()
+            if not settings.DEBUG_QUIETLY:
+                the_narrator.clear_backup()
+            return res
+
+    def __init__(self, duck: Resolvable):
+        self.duck = duck
+
+
+T_duck: TypeAlias = Union[Performable, Answerable, Resolvable]
+T_Qu: TypeAlias = Union[QuietlyAnswerable, QuietlyPerformable, QuietlyResolvable]
+
+
+@overload
+def Quietly(duck: Performable) -> Union[Performable, QuietlyPerformable]:
+    ...
+
+
+@overload
+def Quietly(duck: Answerable) -> Union[Answerable, QuietlyAnswerable]:
+    ...
+
+
+@overload
+def Quietly(duck: Resolvable) -> Union[Resolvable, QuietlyResolvable]:
+    ...
+
+
+@overload
+def Quietly(duck: T) -> T:
+    ...
```

Comment:
Which case is this one for? Is this to suggest folks can pass _anything_ to `Quietly`?

Reply:
Correct.  This shows that what you put in will be what comes out if the other cases fall through.

Reply:
Is that what we want? I feel like we want to tell the user that the `Quietly` won't do anything, maybe with an exception?

Reply:
When the flag is enabled, Quietly does nothing but act as an empty wrapper similar to `given`, `when`, etc.
It made sense to me to give them the ability to add `Quietly` to their code but then turn it off when desired.  
It is possible to resort to always creating a quietly object while only changing the behavior of the methods, but this way we eliminate quietly completely without users resorting to editing their code.

Reply:
So this is the type for when you do `Quietly(SomePerformable())` but our `UNABRIDGED_NARRATION` (name pending) setting is turned on? That makes more sense.

From the tests it seems like you wanted users to be able to do `Quietly(1)` like it's going to do something. From a user perspective, i probably wouldn't think to use it that way, but if i did, i think i'd want `Quietly` to say something like "Can't quiet this one down, please remove me" or something.

Reply:
```python
raise UnableToAct("This duck is too loud!")
```

Reply:
> From the tests it seems like you wanted users to be able to do Quietly(1) like it's going to do something. 

That was unfortunate side effect of testing.  The idea of the test was it technically doesn't matter WHAT you put in Quietly, just that _it_ will be what you get back.  This concept made a bit more sense when answers could have been any object (vs now they should always be Resolvable).  

Though, as I think about the changes we made to resolutions,  there is a strong case for Quietly to only support the 3 types of objects.

Reply:
I think i would prefer to raise an exception if `Quietly` is not able to quiet the object it was given, just to be clear to the user that whatever they expected to happen is not going to. We can make sure to spell that out well in the docs, too.

Reply:
I'm convinced.  See the latest commit